frontend/catalyst/pennylane_extensions.py

```diff
@@ -940,7 +940,7 @@ def _create_jaxpr(init_val, new_cond, new_body):
             new_body, in_tree, init_avals, "while_loop"
         )
         if not treedef_is_leaf(cond_tree) or len(cond_jaxpr.out_avals) != 1:
-            msg = "cond_fun must return a boolean scalar, but got pytree {}."
+            msg = "cond_fun must return a single boolean scalar, but got pytree {}."
             raise TypeError(msg.format(cond_tree))
```

Comment:
Generally we would favour f-strings over the use of `.format`, not a big deal though.

Reply:
Done